File: 10828.py
import sys

class stack:
    def __init__(self):
        self.stack_arr = []
    
    def push(self, num):
        self.stack_arr.append(num)
        
    def pop(self):
        if self.empty() == 1:
            return -1
        else:
            return self.stack_arr.pop()
        
    def size(self):
        return len(self.stack_arr)
        
    def empty(self):
        if self.size() == 0:
            return 1
        else:
            return 0
        
    def top(self):    
        if self.empty() == 1:
            return -1
        else:
            return self.stack_arr[-1]

N = int(sys.stdin.readline())
new_stack = stack()
for i in range(0, N):
    command = sys.stdin.readline().split()
    if command[0] == "push":
        new_stack.push(int(command[1]))
    elif command[0] == "pop":
        res = new_stack.pop()
        print(res)
    elif command[0] == "size":
        res = new_stack.size()
        print(res)
    elif command[0] == "empty":
        res = new_stack.empty()
        print(res)
    elif command[0] == "top":
        res = new_stack.top()
        print(res)

File: 10828.cpp
#include <iostream>
#include <stack>
using namespace std;

class MyStack {
	private:
		stack<int> s;
	
	public:	
		void push(int num) {
			s.push(num);
		}
		
		int pop() {
			if (s.empty()) return -1;
			else {
				int val = s.top();
				s.pop();
				return val;
			}
		}

		int size() {
			return s.size();
		}

		int empty() {
			return s.empty();
		}

		int top() {
			if (s.empty()) return -1;
			else return s.top();
		}
};

int main() {
	MyStack s;
	int N;
	string command;
	int num;

	cin >> N;
	
	for (int i = 0; i < N; i++) {
		cin >> command;

		if (command == "push") {
			cin >> num;
			s.push(num);
		}

		else if (command == "pop") {
			cout << s.pop() << endl;
		}

		else if (command == "size") {
			cout << s.size() << endl;

		}

		else if (command == "empty") {
			cout << s.empty() << endl;

		}

		else if (command == "top") {
			cout << s.top() << endl;

		}
	}
}